scipy/stats/_hypotests.py

```diff
@@ -9,6 +9,223 @@
+
+                # Addresses nan_policy == "omit"
+                if any(contains_nans) and nan_policy == 'omit':
+                    # consider passing in contains_nans
+                    samples = _remove_nans(samples, paired)
+
+                # ideally, this is what the behavior would be, but some
+                # existing functions raise exceptions, so overriding it
+                # would break backward compatibility.
+                # if is_too_small(samples):
+                #     return result_object(np.nan, np.nan)
+
+                return hypotest_fun_in(*samples, *args, **kwds)
+
+            # check for empty input
+            # ideally, move this to the top, but some existing functions raise
+            # exceptions for empty input, so overriding it would break
+            # backward compatibility.
+            empty_output = _check_empty_inputs(samples, axis)
+            if empty_output is not None:
+                statistic = empty_output
+                pvalue = empty_output.copy()
+                return result_object(statistic, pvalue)
+
+            # otherwise, concatenate all samples along axis, remembering where
+            # each separate sample begins
+            lengths = np.array([sample.shape[axis] for sample in samples])
+            split_indices = np.cumsum(lengths)
+            x = scipy.stats.stats._broadcast_concatenate(samples, axis)
+
+            # Addresses nan_policy == "raise"
+            contains_nan, _ = (
+                scipy.stats.stats._contains_nan(x, nan_policy))
+
+            if vectorized and not contains_nan:
+                return hypotest_fun_in(*samples, *args, axis=axis, **kwds)
+
+            # Addresses nan_policy == "omit"
+            if contains_nan and nan_policy == 'omit':
+                def hypotest_fun(x):
+                    samples = np.split(x, split_indices)[:n_samp]
+                    samples = _remove_nans(samples, paired)
+                    if is_too_small(samples):
+                        return result_object(np.nan, np.nan)
+                    return hypotest_fun_in(*samples, *args, **kwds)
+
+            # Addresses nan_policy == "propagate"
+            elif contains_nan and nan_policy == 'propagate':
+                def hypotest_fun(x):
+                    if np.isnan(x).any():
+                        return result_object(np.nan, np.nan)
+                    samples = np.split(x, split_indices)[:n_samp]
+                    return hypotest_fun_in(*samples, *args, **kwds)
+
+            else:
+                def hypotest_fun(x):
+                    samples = np.split(x, split_indices)[:n_samp]
+                    return hypotest_fun_in(*samples, *args, **kwds)
+
+            x = np.moveaxis(x, axis, -1)
+            res = np.apply_along_axis(hypotest_fun, axis=-1, arr=x)
+            return result_object(*result_unpacker(res))
+
+        doc = FunctionDoc(vectorize_hypotest_wrapper)
```

Comment:
Probably should move all this into a function just to keep things a bit cleaner.
Also need to figure out how to fix `source` link in documentation. Note that stats distribution and optimize method-specific documentation suffer from similar problems, so this is nothing new.

Reply:
You mean being able to link the PR `source` link to its code? Because right now only the devdoc is properly linking agreed. Maybe something we could raise on the PyData theme side (we should not do anything here).

Reply:
Do you mean the `[source]` at a page like https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.alpha.html (directly at the right of the signature)? 

I _think_ that's a functionality provided by sphinx itself, not the theme (not fully sure how it is controlled)

Reply:
Yes, that's the link.